### PR TITLE
Utils test update

### PR DESCRIPTION
**Pull request by FJanssen-TNO**
Update to simplify and generalize some energy conservation checks in electricity

### CODE REVIEW

File: tests/test_electric_source_sink.py [hunk 1]
```diff
@@ -46,6 +48,9 @@ def test_source_sink(self):
         results = solution.extract_results()
         parameters = solution.parameters(0)
 
+        # Test energy conservation
+        electric_power_conservation_test(solution, results)
```
Comment (KobusVanRooyen) [inline tests/test_electric_source_sink.py:52]: update docstring: if there are additional items being test in the utils function

File: tests/utils_tests.py [hunk 2]
```diff
@@ -35,23 +34,29 @@ def demand_matching_test(solution, results):
...
+            np.testing.assert_allclose(
+                target, results[f"{d}.Electricity_demand"], atol=1.0e-3, rtol=1.0e-6
+            )
         else:
-            len_times = len(solution.get_timeseries(f"{d}.target_electricity_demand").values)
-        target = solution.get_timeseries(f"{d}.target_electricity_demand").values[0:len_times]
-        np.testing.assert_allclose(
-            target, results[f"{d}.Electricity_demand"], atol=1.0e-3, rtol=1.0e-6
-        )
+            print(f"{d} does not have a target demand profile")
```
Comment (KobusVanRooyen) [inline tests/utils_tests.py:59]: Why is there this print() for electricity but not for the other commodities? -->> "print(f"{d} does not have a target demand profile")"?

File: tests/utils_tests.py [hunk 3]
```diff
@@ -352,6 +357,59 @@ def heat_to_discharge_test(solution, results):
         )
 
 
+def electric_power_conservation_test(solution, results):
+    """
+    Test to check if the electric power is conserved at every timestep
```
Comment (KobusVanRooyen) [inline tests/utils_tests.py:362]: Suggestion? add some high level detail bullet points stating what is checked

File: tests/test_electric_bus.py [hunk 4]
```diff
@@ -46,32 +48,25 @@ def test_voltages_and_power_network1(self):
             input_timeseries_file="timeseries.csv",
         )
         results = solution.extract_results()
+
+        # electric power conservation system and no dissipation of power and current in bus
+        electric_power_conservation_test(solution, results)
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply tests/test_electric_bus.py:53]: This check was already there, but then done explicitly. I removed the manual check and replaced it by this automated one.

File: tests/test_electric_source_sink.py (hunk 1 above)
Comment (FJanssen-TNO) [reply tests/test_electric_source_sink.py:52]: This check was already there, but then done explicitly. I removed the manual check and replaced it by this automated one.

File: tests/test_elec_boiler.py [hunk 5]
```diff
@@ -80,6 +86,7 @@ def test_air_water_hp_elec(self):
         demand_matching_test(heat_problem, results)
         energy_conservation_test(heat_problem, results)
         heat_to_discharge_test(heat_problem, results)
+        electric_power_conservation_test(heat_problem, results)
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply tests/test_elec_boiler.py:89]: updated

File: tests/utils_tests.py (hunk 3 above)
Comment (FJanssen-TNO) [reply tests/utils_tests.py:362]: removed unnecessary in a test.

File: tests/utils_tests.py (hunk 2 above)
Comment (FJanssen-TNO) [reply tests/utils_tests.py:59]: removed unnecessary in a test.

**Review by KobusVanRooyen — APPROVED** [review]

**Pull request merged**